### PR TITLE
[CMSP-675] PHP 8.3 compatibility checks

### PR DESCRIPTION
**Pull request by jazzsequence**
* moves PHPUnit tests from Circle to GitHub Actions
* Adds PHP8.x compatibility checks
* adds wp.org plugin validator
* bumps PHPUnit tests to run on 7.4 and 8.3
* adds PHP linting
* runs PHPCS/PHPLint on 8.3
* adds scheduled cron tests

### CODE REVIEW

File: .github/workflows/lint-test.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
+name: Lint and Test
+
+on:
+  schedule:
+    - cron: '0 0 * * *'
+  push:
+    branches:
+      - '**'
```
Comment (pwtyler) [inline .github/workflows/lint-test.yml]: to clarify, this is to exclude tags? prefer "on pr" IMO.

File: .github/workflows/lint-test.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  test:
+    needs: lint
+    name: Test
+    runs-on: ubuntu-latest
+    services:
+      mariadb:
+        image: mariadb:10.5
+    strategy:
+      matrix:
+        php_version: [7.4, 8.3]
```
Comment (pwtyler) [inline .github/workflows/lint-test.yml:66]:
```suggestion
        php_version: [7.4, 8.2, 8.3]
```
Until 8.3 is GA.

File: bin/install-local-tests.sh
Comment (pwtyler) [inline bin/install-local-tests.sh:1]: feeling like this should be a common subtree somewhere at this point but that's not for here and now.

**Review by pwtyler — APPROVED** [review]
all three comments at your discretion.

File: .github/workflows/lint-test.yml (hunk 2 above)
Comment (jazzsequence) [reply .github/workflows/lint-test.yml:66]: Wouldn't anything that would fail in 8.2 also fail in 8.3, though?

File: bin/install-local-tests.sh
Comment (jazzsequence) [reply bin/install-local-tests.sh:1]: I was thinking the same. could be a composer package maybe? I don't know a better way (other than submodules) to include a dependency package for something like this.

**Pull request merged**